libs/auth-api-lib/src/lib/services/delegations.service.ts

```diff
@@ -86,19 +90,119 @@ export class DelegationsService {
           toNationalId: d.toNationalId,
           fromNationalId: d.fromNationalId,
           fromName: d.fromDisplayName,
-          isFromCompany: d.isFromCompany,
           type: DelegationType.Custom,
           provider: DelegationProvider.Custom,
         },
     )
   }
+
+  async create(
+    nationalId: string,
+    delegation: DelegationDTO,
+  ): Promise<Delegation | null> {
+    this.logger.debug('Creating a new delegation')
+    const id = uuid()
+    await this.delegationModel.create({
+      id: id,
+      fromNationalId: nationalId,
+      ...delegation,
+    })
+    if (delegation.scopes) {
+      this.delegationScopeService.createMany(id, delegation.scopes)
+    }
+    return this.findOne(nationalId, id)
```

Comment:
`this.delegationModel.create` already returns the instance.. you can just return that.